src/MessageEnvelope.php

```diff
@@ -79,6 +79,15 @@ public function incrementRetry()
         return $this;
     }
 
+    public function incrementAndUpdateDate(\DateTime $dateTime = null): self
```

Comment:
Why not simply force the update date in incrementRetry ? Not sure it's usefull to choose the update date and someone can forget to update the date.